### PR TITLE
SI-7938 - parallel collections should use default ExecutionContext

### PR DESCRIPTION
**Pull request by axel22**
Parallel collections now use scala.concurrent ExecutionContext
by default.

Review by @retronym @phaller.

### CODE REVIEW

**Comment by retronym** [comment]
Could you please add a few more code comments in the affected areas? Which tests are exercising which code paths? Is `ExecutionContextProxyTasks` slower, but more general purpose? etc.

I'd also like the the commit comment to explain the background (ie "this was always intended but was forgotten in ..., now we can do this safely because.., the risks of this change are ...")

Thanks!

File: src/library/scala/collection/parallel/Tasks.scala [hunk 1]
```diff
@@ -461,6 +459,53 @@ trait AdaptiveWorkStealingThreadPoolTasks extends ThreadPoolTasks with AdaptiveW
   def newWrappedTask[R, Tp](b: Task[R, Tp]) = new WrappedTask[R, Tp](b)
 }
 
+class ExecutionContextProxyTasks(executor: ExecutionContext) extends Tasks {
```
Comment (retronym) [inline src/library/scala/collection/parallel/Tasks.scala:462]: Should this be part of our public API?

Could it be final?

File: test/files/scalacheck/Ctrie.scala [hunk 2]
```diff
@@ -20,9 +20,9 @@ object Test extends Properties("concurrent.TrieMap") {
 
   /* generators */
 
-  val sizes = choose(0, 200000)
+  val sizes = choose(0, 50000)
 
-  val threadCounts = choose(2, 16)
+  val threadCounts = choose(2, 8)
```
Comment (retronym) [inline test/files/scalacheck/Ctrie.scala:25]: Why did these need changing?

File: src/library/scala/collection/parallel/Tasks.scala (hunk 1 above)
Comment (axel22) [inline src/library/scala/collection/parallel/Tasks.scala:462]: It should be package-private and final. Changed.

File: test/files/scalacheck/Ctrie.scala (hunk 2 above)
Comment (axel22) [inline test/files/scalacheck/Ctrie.scala:25]: Well, I had a feeling that these tests are running slow on my machine, so I changed it, but it seems that's not the case in jenkins.
Either way, can be done in a separate commit - reverting.

**Comment by axel22** [comment]
Following up on your comments, I just noticed some issues - will fix this once again, and resubmit.

**Pull request closed without merging**